diff --git a/src/bson/bson-types.h b/src/bson/bson-types.h
index ca06519..cf62f08 100644
--- a/src/bson/bson-types.h
+++ b/src/bson/bson-types.h
@@ -482,14 +482,14 @@ typedef struct
 BSON_ALIGNED_END (8);
 
 
-BSON_ALIGNED_BEGIN (8)
+//BSON_ALIGNED_BEGIN (8)
 typedef struct _bson_error_t
 {
    uint32_t domain;
    uint32_t code;
    char     message[504];
-} bson_error_t
-BSON_ALIGNED_END (8);
+} bson_error_t;
+//BSON_ALIGNED_END (8);
 
 
 BSON_STATIC_ASSERT (sizeof (bson_error_t) == 512);
