diff --git a/cmake/FindLibBSON.cmake b/cmake/FindLibBSON.cmake
index 90dacc2..107db6b 100644
--- a/cmake/FindLibBSON.cmake
+++ b/cmake/FindLibBSON.cmake
@@ -22,7 +22,7 @@ include(FindPackageHandleStandardArgs)
 # Load up PkgConfig if we have it
 find_package(PkgConfig QUIET)
 
-if (PKG_CONFIG_FOUND)
+if (LIBBSON_DIR STREQUAL "" AND PKG_CONFIG_FOUND)
   pkg_check_modules(LIBBSON REQUIRED libbson-1.0>=${LibBSON_FIND_VERSION} )
   # We don't reiterate the version information here because we assume that
   # pkg_check_modules has honored our request.
