diff --git a/cmake/FindLibBSON.cmake b/cmake/FindLibBSON.cmake
index 90dacc2..107db6b 100644
--- a/cmake/FindLibBSON.cmake
+++ b/cmake/FindLibBSON.cmake
@@ -22,7 +22,7 @@ include(FindPackageHandleStandardArgs)
 # Load up PkgConfig if we have it
 find_package(PkgConfig QUIET)
 
-if (PKG_CONFIG_FOUND)
+if (LIBBSON_DIR STREQUAL "" AND PKG_CONFIG_FOUND)
   pkg_check_modules(LIBBSON REQUIRED libbson-1.0>=${LibBSON_FIND_VERSION} )
   # We don't reiterate the version information here because we assume that
   # pkg_check_modules has honored our request.
diff --git a/cmake/FindLibMongoC.cmake b/cmake/FindLibMongoC.cmake
index c226188..30023a7 100644
--- a/cmake/FindLibMongoC.cmake
+++ b/cmake/FindLibMongoC.cmake
@@ -22,7 +22,7 @@ include(FindPackageHandleStandardArgs)
 # Load up PkgConfig if we have it
 find_package(PkgConfig QUIET)
 
-if (PKG_CONFIG_FOUND)
+if (LIBMONGOC_DIR STREQUAL "" AND PKG_CONFIG_FOUND)
   pkg_check_modules(LIBMONGOC REQUIRED libmongoc-1.0>=${LibMongoC_FIND_VERSION} )
   # We don't reiterate the version information here because we assume that
   # pkg_check_modules has honored our request.
@@ -31,7 +31,7 @@ elseif(LIBMONGOC_DIR)
   # The best we can do until libMONGOC starts installing a libmongoc-config.cmake file
   set(LIBMONGOC_LIBRARIES mongoc-1.0 CACHE INTERNAL "")
   set(LIBMONGOC_LIBRARY_DIRS ${LIBMONGOC_DIR}/lib CACHE INTERNAL "")
-  set(LIBMONGOC_INCLUDE_DIRS ${LIBMONGOC_DIR}/include/libMONGOC-1.0 CACHE INTERNAL "")
+  set(LIBMONGOC_INCLUDE_DIRS ${LIBMONGOC_DIR}/include/libmongoc-1.0 CACHE INTERNAL "")
   find_package_handle_standard_args(LIBMONGOC DEFAULT_MSG LIBMONGOC_LIBRARIES LIBMONGOC_LIBRARY_DIRS LIBMONGOC_INCLUDE_DIRS)
 else()
     message(FATAL_ERROR "Don't know how to find libmongoc; please set LIBMONGOC_DIR to the prefix directory with which libbson was configured.")
